EXTRA_FIM/pyiron_job.py

```diff
@@ -0,0 +1,247 @@
+        self.input["kpoint"] = None
+        self.input["ionization_energies"] = None
+        self.input["extrapolate_potential"] = False
+        self.input["total_kpoints"] = None
+
+    def extrpolate_potential(self):
+        """Extrapolate potential if needed, to extrapolate waves to higher distances"""
+
+        elec_potential, _ = sx_el_potential3D_cell(
+            self.input["simulator_dict"]["working_directory"]
+        )
+        pot, _, _, cell = fim.potential(self.input["simulator_dict"]).potential_cell()
+
+        if self.input.extrapolate_potential:
+            iz0 = self.input["simulator_dict"]["iz_ext_from"]
+            new_z_max = self.input["simulator_dict"]["z_ext"]
+            _, pot_ext = extend_potential(
+                elec_potential / fim.HARTREE_TO_EV,
+                iz0,
+                pot,
+                cell,
+                z_max=new_z_max,
+                izend=self.input["simulator_dict"]["izend"],
+                dv_limit=1e-4,
+                plotG=1,
+            )
+            # copy extension from pot to elec_potential
+            elec_ext = pot_ext[:, :, :, 0] * fim.HARTREE_TO_EV
+            elec_ext[:, :, 0:iz0] = elec_potential[:, :, :iz0]
+        else:
+            pot_ext = pot
+            elec_ext = elec_potential
+        return pot_ext, elec_potential
+
+    @property
+    def suggest_input_dict(self):
+        """Suggests a input dictionary based on the electrostatic potential, Fermi and ionization energy"""
+        waves_reader = sx_nc_waves_reader(
+            Path(self.input["waves_directory"]) / "waves.sxb"
+        )
+        e_fermi = waves_reader.get_fermi_energy()
+        _, sim = suggest_input_dictionary(
+            self.input["waves_directory"],
+            e_fermi,
+            ionization_energies=self.input["ionization_energies"],
+        )
+        self.input["total_kpoints"] = waves_reader.nk
+        self.input["simulator_dict"] = sim
+        self.input["z_max"] = sim["z_max"]  # rename later
+        self.input["izstart_min"] = sim["izstart_min"]  # rename later
+        self.input["izend"] = sim["izend"]  # rename later
+        self.input["limit"] = sim["limit"]  # rename later
+        self.input["cutoff"] = sim["cutoff"]  # rename later
+        self.input["E_fermi"] = sim["E_fermi"]  # rename later
+        self.input["E_max"] = sim["E_max"]  # rename later
+        return sim
+
+    def run_static(self):
+
+        self.project_hdf5.create_working_directory()
+        # self.suggest_input_dict
+        pot_ext, elec_ext = self.extrpolate_potential()
+        waves_reader = sx_nc_waves_reader(self.input["waves_directory"] + "/waves.sxb")
+        fimsim = fim.FIM_simulations(
+            self.input["simulator_dict"],
+            reader=waves_reader,
+            V_total=pot_ext,
+            V_elstat=elec_ext,
+        )
+        self.status.submitted = True
+        fimsim.sum_single_k(self.input["kpoint"], path=self.working_directory)
```

Comment:
See above. I suggest to move hd5 writing out of sum_single_k function and do it here explicitly (where we know that and where we want to write).